ceci/stage.py

```diff
@@ -168,19 +242,22 @@ def __init_subclass__(cls, **kwargs):
         filename = sys.modules[cls.__module__].__file__
 
         stage_is_complete = (
-            hasattr(cls, "outputs") and hasattr(cls, "inputs") and hasattr(cls, "run")
+            hasattr(cls, 'inputs') and hasattr(cls, 'outputs') and not getattr(cls.run, '__isabstractmethod__', False)
```

Comment:
Right now the hasattrs will always be True because the variables are defined in the base class.

Reply:
Fixed, as above